sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -256,33 +257,78 @@ export class AssessmentControllerService {
 
   /**
    * Populate the final awards in a dynamic way like disbursement schedule(estimated) awards.
-   * @param disbursementReceipts disbursement receipt details.
-   * @param disbursementSchedule disbursement schedule of the disbursement receipt(s).
-   * @param identifier identifier which is used to create dynamic data by appending award code
-   * to it.
+   * @param assessment student assessment.
+   * @param options options,
+   * - `studentId` studentId student to whom the award details belong to.
+   * - `applicationId` application is used for authorization purposes to
+   * ensure that a user has access to the specific application data.
    * @returns dynamic award data of disbursement receipts of a given disbursement.
-   * @example
-   * finalAward: {
-   *    disbursementReceipt1cslf: 1000,
-   *    disbursementReceipt1csgp: 10001,
-   *    disbursementReceipt1bcsl: 1005,
-   *    disbursementReceipt1bcag: 1006,
-   *    disbursementReceipt1bgpd: 1007,
-   *    disbursementReceipt1sbsd: 1008,
-   *    disbursementReceipt2cslf: 1000,
-   *    disbursementReceipt2csgp: 10001,
-   *    disbursementReceipt2bcsl: 1005,
-   *    disbursementReceipt2bcag: 1006,
-   *    disbursementReceipt2bgpd: 1007,
-   *    disbursementReceipt2sbsd: 1008,
+   */
+  private async populateDisbursementFinalAwardsValues(
+    assessment: StudentAssessment,
+    options?: {
+      studentId?: number;
+      applicationId?: number;
+    },
+  ): Promise<DynamicAwardValue | undefined> {
```

Comment:
why cant we add the undefined in the DynamicAwardValue?

Reply:
We can. Adding it to the possible return type is just to make it explicit.